src/core/thread/mle_router.cpp

```diff
@@ -1369,9 +1361,15 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
 
                     break;
                 }
+
+                if (mRouterSelectionJitterTimeout == 0 &&
```

Comment:
thanks. will recover back and give another proposal to solve the special scenario I met.